### PR TITLE
Update retokenization tool to support RoBERTa; fix WSC

### PR DESCRIPTION
**Pull request by sleepinyourhat**
It looks like the retokenization code wasn't touched in #890, which means that WSC isn't supported with RoBERTa models. This should fix that.

### CODE REVIEW

File: jiant/tasks/qa.py [hunk 1]
```diff
@@ -273,10 +273,12 @@ def tokenize_preserve_placeholder(sent):
...
             sent_parts = sent.split("@placeholder")
             assert len(sent_parts) == 2
-            sent_parts = [
-                tokenize_and_truncate(self.tokenizer_name, s, self.max_seq_len) for s in sent_parts
-            ]
-            return sent_parts[0] + ["@placeholder"] + sent_parts[1]
+            placeholder_loc = len(
+                tokenize_and_truncate(self.tokenizer_name, sent_parts[0], self.max_seq_len)
+            )
+            sent_tok = tokenize_and_truncate(self.tokenizer_name, sent, self.max_seq_len)
```
Comment (iftenney) [inline jiant/tasks/qa.py]: This seems a little sketchy - prefer something like `return sent_tok[:i] + ["@placeholder"] + sent_tok[i:]` than mutating with insert?

Also, should this be truncated to `self.max_seq_len - ` to account for the placeholder?

File: jiant/utils/retokenize.py [hunk 2]
```diff
@@ -328,12 +339,48 @@ def align_wpm(text: Text, tokenizer_name: str) -> Tuple[TokenAligner, List[Text]
...
+    sentencepiece_tokens = sentencepiece_tokenizer.tokenize(text)
+
+    modified_sentencepiece_tokens = list(
+        map(process_sentencepiece_for_alignment, sentencepiece_tokens)
+    )
+    ta = TokenAligner(bow_tokens, modified_sentencepiece_tokens)
+    return ta, sentencepiece_tokens
+
+
+def align_bpe(text: Text, tokenizer_name: str) -> Tuple[TokenAligner, List[Text]]:
```
Comment (iftenney) [inline jiant/utils/retokenize.py]: Add function docstrings giving examples and stating which models these apply to?

Since these are fairly dense and involve a number of edge cases, should we write some unit tests for this module?

File: jiant/tasks/qa.py (hunk 1 above)
Comment (HaokunLiu) [reply jiant/tasks/qa.py]: Thanks.

File: jiant/utils/retokenize.py (hunk 2 above)
Comment (HaokunLiu) [reply jiant/utils/retokenize.py]: I am afraid it will not be very economical to do this.
The only occasions we need to modify this, is either adding new tokenizer or refactoring retokenize. When the first happens, the old test cases can always pass, and we always need to add new test cases (which has little reuse value). When the second happens, I don't expect we will keep the same interface.
I wonder what do you think about it?

Comment (iftenney) [reply jiant/utils/retokenize.py]: I don't think it'll be hard - this module has a very minimal API so it should be easy to write tests. We are refactoring and adding new tokenizers here, and we'd want tests both to ensure there aren't regressions and that the new functionality is doing what we expect.

(This is partly my fault for not having any on the original version, but since it's grown quite a bit in scope and tokenization bugs can be very hard to detect otherwise, I think it's warranted now.)

Comment (HaokunLiu) [reply jiant/utils/retokenize.py]: I see. I'll do it later today.

**Comment by sleepinyourhat** [comment]
What's left to do here? We've been holding off on the RoBERTa release for a while now...

**Comment by HaokunLiu** [comment]
> What's left to do here? We've been holding off on the RoBERTa release for a while now...

I have added docstrings and test cases, the only problem is using bypebpe tokenizer requires download spacy en, I don't know if there is anyway to do it in circleci. If we simply remove the test case for bytebpe, then this PR is ready to go. Otherwise I'm still waiting to see if @davidbenton  can tell me something.

**Comment by sleepinyourhat** [comment]
Ah, that should be easy—I didn't realize you were waiting. CircleCI just runs a list of commands, so you can just add a shell command to download the data here: 

https://github.com/nyu-mll/jiant/blob/master/.circleci/config.yml#L50

If it takes more than a few minutes to download, though, it's okay to skip.

**Comment by HaokunLiu** [comment]
This should be ready now.

**Comment by sleepinyourhat** [comment]
Great! @iftenney, merge when ready.

File: jiant/utils/retokenize.py [hunk 3]
```diff
@@ -294,28 +297,40 @@ def process_wordpiece_for_alignment(t):
...
+    if t.startswith("▁"):
+        return "<w>" + re.sub(r"^▁", "", t)
+    else:
+        return t
+
+
+def process_bytebpe_for_alignment(t):
+    """Add <w> markers to ensure word-boundary alignment."""
+    if t.startswith("▁"):
+        return "<w>" + re.sub(r"^Ġ", "", t)
```
Comment (iftenney) [inline jiant/utils/retokenize.py]: Is "^Ġ" an escape sequence for something, or is this the evil-underscore?
